cli/utils/index.js

```diff
@@ -6,15 +6,27 @@ const config = require('../config')
 const DATA_FILE_PATH = path.join(__dirname, '../', 'data')
 
 function saveToken (token) {
-  // TODO: save the token somewhere
+  return fs.writeFileSync(DATA_FILE_PATH, token)
 }
 
 function readToken () {
-  // TODO: read the token from the saved location
+  if (!canUserAccessDataFile()) return null
+  return fs.readFileSync(DATA_FILE_PATH)
 }
 
 function deleteToken () {
-  // TODO: delete the currently saved token
+  if (!canUserAccessDataFile()) return
+  return fs.unlinkSync(DATA_FILE_PATH)
+}
+
+function canUserAccessDataFile () {
+  try {
+    // accessSync will return undefined on success and throw on failure
+    fs.accessSync(DATA_FILE_PATH)
+    return true
+  } catch (e) {
+    return false
+  }
```

Comment:
This function is where we check if the file actually exists. This fixes the bug that we encountered during the session.
I tried using the same solution during the session but forgot that `access` and `accessSync` throw errors when the file cannot be accessed. I assumed that they work with boolean values during the session. I was thinking of `existsSync`. Apologies!